test-app/runtime/src/main/cpp/ObjectManager.cpp

```diff
@@ -341,6 +341,10 @@ void ObjectManager::JSObjectFinalizer(Isolate *isolate, ObjectWeakCallbackState
         auto jsInfoIdx = static_cast<int>(MetadataNodeKeys::JsInfo);
         po->Get(m_isolate)->SetInternalField(jsInfoIdx, Undefined(m_isolate));
         po->Reset();
+        auto it = m_idToObject.find(javaObjectID);
+        m_idToObject.erase(it);
+        delete po;
+        delete callbackState;
     }
```

Comment:
Also delete the `po` and `callbackState` variables here: https://github.com/NativeScript/android-runtime/pull/1566/files#diff-6c6c152bfa906bd1defa04a2daacdedbR327

Reply:
Good point, thanks, we'll add that.